tools/include/operation.hpp

```diff
@@ -141,7 +141,8 @@ auto compute_shape_op(rank<2>, const T& x, const std::vector<shape>& inputs)
     -> decltype(x.normalize_compute_shape(inputs))
 {
     dependent_type<operation, T> y = x;
-    normalize_attributes(y, inputs[0].max_lens());
+    if(inputs.size() > 0)
+        normalize_attributes(y, inputs[0].max_lens());
```

Comment:
I'm not sure if we should be allowing this to continue to normalize_compute_shape() or throwing here. @pfultz2

Reply:
The one op. anyone identified that takes 0 arguments is Allocate--it may be allowed to continue.  Its compute_shape returns a default shape.

Reply:
right, but allocate doesn't use normalize_attributes.

Reply:
Paul and I agree that it should instead throw here in operation.hpp if there are no inputs. This will make your atleast one check_shapes not pass code coverage, so just remove the check for now.